opacus/privacy_engine.py

```diff
@@ -56,6 +56,15 @@ def validate(
         """
         ModuleValidator.validate(module, raise_if_error=True)
 
+    def get_compatible_module(self, module: nn.Module) -> nn.Module:
```

Comment:
>We can make it a @classmethod.

I don't see why not. Makes more sense since it doesn't depend on any instance attributes.

>what's the downside of asking people to call ModelValidator.fix() directly?

1. Users having to import `ModuleValidator` in their training file (in addition to PrivacyEngine).
2. We lose the [single-window system ](https://www.wikiwand.com/en/Single-window_system) for accessing Opacus internals.

Reply:
>Users having to import ModuleValidator in their training file

That make sense. I don't have a strong preference, and this argument alone is enough to justify this decision I think.

> We lose the single-window system for accessing Opacus internals.

As a more general point, I'd love to have a discussion about this at some point. I would argue that having a single-window system is not the most fitting design approach, despite opacus historically being this way.